src/QMCHamiltonians/ECPComponentBuilder.2.cpp

```diff
@@ -49,17 +49,66 @@ void ECPComponentBuilder::buildSemiLocalAndLocal(std::vector<xmlNodePtr>& semiPt
   int nso   = 0;
   Llocal    = -1;
   OhmmsAttributeSet aAttrib;
+  int quadRule = -1;
   aAttrib.add(eunits, "units");
   aAttrib.add(format, "format");
   aAttrib.add(ndown, "npots-down");
   aAttrib.add(nup, "npots-up");
   aAttrib.add(Llocal, "l-local");
-  aAttrib.add(Nrule, "nrule");
+  aAttrib.add(quadRule, "nrule");
   aAttrib.add(Srule, "srule");
   aAttrib.add(nso, "npots-so");
 
   xmlNodePtr cur_semilocal = semiPtr[0];
   aAttrib.put(cur_semilocal);
+
+  if (quadRule > -1 && Nrule > -1)
+  {
+    app_warning() << " Nrule setting found in both qmcpack input and pseudopotential file."
+                  << " Using nrule setting in qmcpack input file\n";
+  }
+  else if (quadRule > -1 && Nrule == -1)
+  {
+    app_log() << " Nrule setting found in pseudopotential file.\n";
```

Comment:
"Nrule setting found in pseudopotential file and used."
std::endl incluses a flush. it is preferred to "\n"